includes/Listeners/WonderCart.php

```diff
@@ -13,32 +13,155 @@ class WonderCart extends Listener {
 	 * @return void
 	 */
 	public function register_hooks() {
-		add_action('rest_after_insert_yith_campaign', array( $this, 'register_campaign' ), 10 );
+		add_action( 'rest_after_insert_yith_campaign', array( $this, 'register_campaign' ), 10 );
+		add_action( 'yith_sales_edit_campaign_event_modal_opened', array( $this, 'create_campaign_modal_open' ), 10, 2 );
+		add_action( 'yith_sales_edit_campaign_event_campaign_selected', array( $this, 'campaign_selected' ), 10, 2 );
+		add_action( 'yith_sales_edit_campaign_event_campaign_abandoned', array( $this, 'campaign_abandoned' ), 10, 2 );
+		add_action( 'woocommerce_before_checkout_form', array( $this, 'checkout_campaigns_used' ) );
```

Comment:
@mamatharao05 We have to be very careful with front-end actions. The last time we enabled a checkout-related action, we incurred significant costs.

Reply:
@wpscholar As per the discussion with Kasturi, Updated the event to trigger when the order is processed and it will only be triggered if the campaigns are used.